src/smtp.rs

```diff
@@ -48,6 +48,16 @@ impl Smtp {
 
     /// Connect using the provided login params
     pub fn connect(&mut self, context: &Context, lp: *const dc_loginparam_t) -> usize {
+        warn!(context, 0, "SMTP ***************** CONNECT");
+        unsafe {
+            dc_log_event(
```

Comment:
IIRC you can do `info!(context, 0, "SMTP *** connect starts")` here, and then there is no need for the unsafe block.  (if that's wrong search for an actual example for the signature)

Reply:
done :)